airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/cdc/MongoDbCdcConnectorMetadataInjector.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mongodb.internal.cdc;
+
+import static io.airbyte.integrations.debezium.internals.DebeziumEventUtils.CDC_DELETED_AT;
+import static io.airbyte.integrations.debezium.internals.DebeziumEventUtils.CDC_UPDATED_AT;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airbyte.integrations.debezium.CdcMetadataInjector;
+import io.airbyte.integrations.debezium.internals.DebeziumEventUtils;
+import io.airbyte.integrations.debezium.internals.mongodb.MongoDbDebeziumConstants;
+import io.airbyte.integrations.debezium.internals.mongodb.MongoDbResumeTokenHelper;
+import org.bson.BsonTimestamp;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * MongoDB specific implementation of the {@link CdcMetadataInjector} that stores the MongoDB resume
+ * token timestamp in the {@link #CDC_UPDATED_AT} metadata field.
+ */
+public class MongoDbCdcConnectorMetadataInjector implements CdcMetadataInjector<BsonTimestamp> {
+
+  static final String CDC_DEFAULT_CURSOR = "_ab_cdc_cursor";
+
+  private final long emittedAtConverted;
+
+  // This now makes this class stateful. Please make sure to use the same instance within a sync
+  private final AtomicLong recordCounter = new AtomicLong(1);
+
+  public MongoDbCdcConnectorMetadataInjector(long emittedAtConverted) {
+    this.emittedAtConverted = emittedAtConverted;
+  }
+
+  @Override
+  public void addMetaData(final ObjectNode event, final JsonNode source) {
+    final BsonTimestamp timestamp = MongoDbResumeTokenHelper.extractTimestampFromSource(source);
+    event.put(CDC_UPDATED_AT, timestamp.getValue());
```

Comment:
We are already adding `base.put(CDC_UPDATED_AT, transactionTimestamp);` This is redundant right?

Reply:
@subodh1810 I didn't realize that the `event` passed to `addMetaData` contains the metadata from the other method.  If that is the case, then it is redundant, though I think we then need to talk about which value is the correct one.